src/app/components/reports/pages/event-summary/EventSummaryReportParameters.ts

```diff
@@ -0,0 +1,174 @@
+import { toBase64Url } from "@helpers/encoding/encoding";
+import { filterDate, filterTime } from "@helpers/filters/audioRecordingFilters";
+import { filterAnd, filterModelIds } from "@helpers/filters/filters";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { Id } from "@interfaces/apiInterfaces";
+import { EventSummaryReport } from "@models/EventSummaryReport";
+import { DateTime, Duration } from "luxon";
+
+export enum ChartType {
+  sensorPointMap = "Sensor Point Map",
+  speciesAccumulationCurve = "Species Accumulation Curve",
+  speciesCompositionCurve = "Species Composition Curve",
+  falseColorSpectrograms = "False Colour Spectrograms",
+}
+
+export enum BinSize {
+  day = "day",
+  week = "week",
+  fortnight = "fortnight",
+  month = "month",
+  season = "season",
+  year = "year",
+}
+
+export interface IEventSummaryReportParameters {
+  sites: Id[];
+  points: Id[];
+  provenances: Id[];
+  events: Id[];
+  recogniserCutOff: number;
+  charts: string[];
+  binSize: BinSize;
+  ignoreDaylightSavings: boolean;
+}
+
+export class EventSummaryReportParameters
+  implements IEventSummaryReportParameters
+{
+  public constructor(queryStringParameters: Params = {}) {
+    const conversionTable = {
+      sites: (value: string) => value.split(",").map(Number),
+      points: (value: string) => value.split(",").map(Number),
+      provenances: (value: string) => value.split(",").map(Number),
+      events: (value: string) => value.split(",").map(Number),
+      recogniserCutOff: (value: string) => Number(value),
+      charts: (value: string) => value.split(","),
+      binSize: (value: string) => value,
+      ignoreDaylightSavings: (value: string) => value === "true",
+      dateStartedAfter: (value: string) => DateTime.fromISO(value, { zone: "utc" }),
+      dateFinishedBefore: (value: string) => DateTime.fromISO(value, { zone: "utc" }),
+      timeStartedAfter: (value: string) => Duration.fromObject({
+        hours: Number(value.split(":")[0]),
+        minutes: Number(value.split(":")[1]),
+      }),
+      timeFinishedBefore: (value: string) => Duration.fromObject({
+        hours: Number(value.split(":")[0]),
+        minutes: Number(value.split(":")[1]),
+      }),
+    };
+
+    // since query string parameters are losely typed using a string from the user space
+    // we have to use this hacky implementation of manual key-value checking and assignment
+    Object.entries(queryStringParameters).forEach(([key, value]) => {
+      if (key in conversionTable) {
+        this[key] = conversionTable[key](value);
+      }
+    });
+  }
+
+  // since these properties are exposed to the user in the form of query string parameters
+  // we use the user friendly names
+  public sites: Id[];
+  public points: Id[];
+  public provenances: Id[];
+  public events: Id[];
+  public recogniserCutOff = 0;
+  public charts: string[];
+  public binSize: BinSize = BinSize.month;
+  public ignoreDaylightSavings = true;
```

Comment:
I'll be using `daylightSavings` only because our [user facing fields conventions](https://github.com/QutEcoacoustics/workbench-client/blob/86eded2efcabbdc5ad008cd747cd73d9396bfe4b/CONTRIBUTING.md?plain=1#L19-L26) _slightly_ cover this scenario by stating that abbreviations aren't used for user facing fields